core/common/src/main/java/alluxio/resource/CloseableIterator.java

```diff
@@ -135,4 +135,15 @@ public void closeResource() {
       }
     };
   }
+
+  /**
+   * Consumes the iterator, closes it, and returns its size.
+   * @param iter the iter to get the size of
+   * @return the size of the iter
+   */
+  public static int size(CloseableIterator<?> iter) {
+    int size = Iterators.size(iter);
+    iter.close();
+    return size;
+  }
```

Comment:
is there a better way than consuming this iter just to see the size? I'd guess if the iterator really can't give you the size, you should probably look elsewhere other than brute forcing this iterator?

Reply:
This is just used in tests to check that there are the correct number of children. I can move this function somewhere to make sure it is only used in tests.

Reply:
SG, if you could move it out then perfect. Otherwise I guess you can annotate with sth like `@VisibleForTesting` ?